sdk/core/azure-core-http-okhttp/src/main/java/com/azure/core/http/okhttp/OkHttpAsyncHttpClientBuilder.java

```diff
@@ -0,0 +1,189 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.core.http.okhttp;
+
+import com.azure.core.http.HttpClient;
+import com.azure.core.util.logging.ClientLogger;
+import okhttp3.Authenticator;
+import okhttp3.ConnectionPool;
+import okhttp3.Dispatcher;
+import okhttp3.Interceptor;
+import okhttp3.OkHttpClient;
+
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+
+/**
+ * Builder to configure and build an implementation of com.azure.core.http.HttpClient for OkHttp.
+ */
+public class OkHttpAsyncHttpClientBuilder {
```

Comment:
Should the builder just be named as `OkHttpClientBuilder` instead of adding `Async`. Other client builders don't add `Async` to the classname.

Reply:
Not sure about removing "Async" word from builder name, I followed reactory-netty HttpClient builder naming convention. 

Also the naming pattern we follow is: `[productName][Async]HttpClientBuilder` where product name is `Netty`, `OkHttp`. 

With 'Async' removed, the pattern is `[productName]HttpClientBuilder` so `OkHttpHttpClientBuilder`, the word Http repeats. If change it to `OkHttpClientBuilder` as suggested then it is no longer following the pattern. 

@JonathanGiles thoughts?